src/components/describe/describeWebview.ts

```diff
@@ -1,23 +1,51 @@
 import * as vscode from 'vscode';
+import { Kubectl } from '../../kubectl';
+import { ShellResult } from '../../shell';
 import { WebPanel } from '../webpanel/webpanel';
 
 export class DescribePanel extends WebPanel {
     public static readonly viewType = 'vscodeKubernetesDescribe';
     public static currentPanels = new Map<string, DescribePanel>();
 
-    public static createOrShow(content: string, resource: string) {
+    public static createOrShow(content: string, resource: string, cmd: string, kubectl: Kubectl) {
```

Comment:
Style suggestion: instead of passing in these objects and asking the view to compose the kubectl invocation, consider passing in a `refreshCallback: () => Promise<Errorable<string>>` (or `Promise<ShellResult | undefined>` if you prefer).  Having the mechanics of description retrieval split across this module and its caller (for the original content) feels a bit awkward to me.

Alternatively, you could make this module _entirely_ responsible for populating the description - so it would use `cmd` and `kubectl` to compose the initial content as well as the refreshed content.  You might even be able to use the same mechanism to inject both the initial content and the refreshed content into the view.

Reply:
done.